components/devfile-page/DevPageHeader.tsx

```diff
@@ -0,0 +1,116 @@
+  devfileMetadata?: DevfileMetadata;
+}
+
+/**
+ * header that displays basic information and metadata information for respective devfile
+ * @param devfile - index information for devfie
+ * @param metadata - metadata information from devfile yaml
+ */
+const DevPageHeader = ({ devfile, devfileMetadata }: Props) => {
+  const devfileMetaInclude = ['projectType', 'version', 'language']; // types to include in metadata from index
+
+  return (
+    <Card
+      isRounded
+      style={{
+        width: '75%',
+        maxWidth: '1000px',
+        margin: 'auto',
+        marginTop: '30px',
+        marginBottom: '30px',
+      }}
+    >
+      <CardBody style={{ width: '100%' }}>
+        <div style={{ display: 'flex', width: '100%' }}>
+          <div
+            style={{
+              display: 'flex',
+              width: '50%',
+              borderRight: '2px solid #ADABAE',
+            }}
+          >
+            <div style={{ margin: '10px' }}>
+              <Brand
+                src={devfile.icon || devfileLogo}
+                alt={devfile.icon ? devfile.name + ' logo' : 'devfile logo'}
+                style={{ width: '60%' }}
+              />
+              <Text
+                style={{
+                  color: '#ADABAE',
+                  marginLeft: 'auto',
+                  marginRight: 'auto',
+                }}
+              >
+                {devfile.type}
+              </Text>
+            </div>
+            <div style={{ margin: '10px' }}>
+              <Text style={{ fontSize: '22px', margin: '0.5rem' }}>
+                {devfile.displayName}
+              </Text>
+              <Text style={{ margin: '0.5rem' }}>
+                &emsp;{devfile.description}
+              </Text>
+              <HeaderTags tags={devfile.tags} />
+            </div>
+          </div>
+          <div style={{ width: '50%', margin: '2%' }}>
+            {Object.entries(devfile).map(([key, value]) => {
+              if (devfileMetaInclude.includes(key)) {
+                let label = key.replace(/([a-z](?=[A-Z]))/g, '$1 '); // split camel case up
+                label = label[0].toUpperCase() + label.substring(1);
+                return (
+                  <Text>
+                    <strong>{label + ': '}</strong>
+                    {value}
+                  </Text>
+                );
+              }
+            })}
+            {devfile.type === 'stack' // include website if stack; include git if sample
+              ? devfileMetadata &&
+                devfileMetadata.website && (
+                  <Text>
+                    <strong>Website: </strong>
+                    <a
+                      href={devfileMetadata.website}
+                      target="_blank"
+                      rel="noreferrer"
+                    >
+                      {devfileMetadata.website}
+                    </a>
+                  </Text>
+                )
+              : devfile.git && (
+                  <a
```

Comment:
You should wrap `<a>` with `<Link>` from `import Link from 'next/link';`
Look at https://github.com/devfile/registry-viewer/blob/736bb0d4b0f9c18fd43d1ef33496b2f7d567bdce/components/page/Layout.tsx#L45 for an example. You only need to move the `href` inside of the `<Link>` This is done for precaching purposes on Nextjs.

Reply:
Make sure to do this everywhere where you have an anchor tag that has an href.

Reply:
even for links leading outside the registry viewer ?

Reply:
Yes

Reply:
https://github.com/vercel/next.js/blob/master/errors/invalid-href-passed.md 
according to their documentation, next/link should only be used for internal navigation and not external navigation